clang/include/clang/Basic/Attr.td

```diff
@@ -1135,6 +1135,28 @@ def SYCLKernel : InheritableAttr {
   let Documentation = [SYCLKernelDocs];
 }
 
+// Marks functions which must not be vectorized via horizontal SIMT widening,
+// e.g. because the function is already vectorized. Used to mark SYCL
+// explicit SIMD kernels and functions.
+def SYCLSimd : InheritableAttr {
+  let Spellings = [GNU<"sycl_explicit_simd">];
+  let Subjects = SubjectList<[Function]>;
+  let LangOpts = [SYCLIsDevice];
```

Comment:
But I just realized, should LangOpts value, be `SYCLExplicitSIMD` instead of `SYCLIsDevice`?

Reply:
right